src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -125,7 +125,7 @@ public function handle(AddProductToOrderCommand $command)
             $cart,
             $order->getCurrentOrderState(),
             $cart->getProducts(),
-            $command->getOrderInvoiceId() ?: 0
+            $invoice->id ?: 0
```

Comment:
Mmm, what is wrong with `$command->getOrderInvoiceId()` it should contain the correct id no?

Reply:
That was the bug:

$command->getOrderInvoiceId() works fine when we are adding a product, modifying an existing invoice, the command does have an InvoiceId. But if we are creating a new invoice for this added product, then the invoice was just created and is not present in the command. Using the $invoice object works in both situations.

Reply:
oohhh, smart!